From 87d0d4f3a0307937a8c9ca7bc2f8150eca06b181 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Fri, 9 Jan 2015 14:19:36 +0530
Subject: [PATCH 2/3] pmic_ccsm: Set SDP current based on usb compliance bit

If USB compliance bit is set, then set current as 100mA else set to
500mA.

Change-Id: Ie1fa115a1ee72de6280f4f1db6522679891669ee
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index b6dd76f..b3aff49 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -879,7 +879,10 @@ static void handle_internal_usbphy_notifications(int mask)
 			evt =  USB_EVENT_VBUS;
 		else
 			evt =  USB_EVENT_NONE;
-		cap.ma = LOW_POWER_CHRG_CURRENT;
+		if (chc.pdata->usb_compliance)
+			cap.ma = USBINPUTICC100VAL;
+		else
+			cap.ma = LOW_POWER_CHRG_CURRENT;
 		break;
 	case POWER_SUPPLY_CHARGER_TYPE_USB_CDP:
 		if (cap.chrg_evt == POWER_SUPPLY_CHARGER_EVENT_CONNECT)
-- 
1.9.1

